File: src/main/java/org/git/desk/component/FileMenu.java
package org.git.desk.component;

import javafx.event.ActionEvent;
import javafx.scene.Scene;
import javafx.scene.control.Menu;
import javafx.scene.control.MenuItem;
import javafx.stage.Stage;
import org.git.desk.loader.FXMLLoaderFactory;
import org.kordamp.ikonli.javafx.FontIcon;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class FileMenu extends Menu {

  private static final Logger log = LoggerFactory.getLogger(FileMenu.class);

  private final MenuItem addAccount;
  private final MenuItem addRepository;

  public FileMenu() {
    super("_File", new FontIcon("mdi2f-folder"));

    // ───── 初始化菜单项 ─────
    addAccount = new IconMenuItem("Add Account", "mdi2f-folder-account");
    addRepository = new IconMenuItem("Add Repository", "mdi2s-source-repository"); // 可换图标

    // ───── 设置事件 ─────
    addAccount.setOnAction((ActionEvent event) -> openForm("AddAccountForm.fxml", "Add Account"));
    addRepository.setOnAction((ActionEvent event) -> openForm("AddRepositoryForm.fxml", "Add Repository"));

    // ───── 添加到菜单 ─────
    this.getItems().addAll(addAccount, addRepository);
  }

  // ───── 打开表单窗口 ─────
  private void openForm(String fxmlPath, String title) {
    try {
      var root = FXMLLoaderFactory.load(fxmlPath);
      Stage stage = new Stage();
      stage.setTitle(title);
      stage.setScene(new Scene(root));
      stage.show();
    } catch (Exception e) {
      log.error("Failed to open form: {}", fxmlPath, e);
    }
  }
}

File: src/main/kotlin/org/git/desk/component/FileMenu.kt
package org.git.desk.component

import io.github.oshai.kotlinlogging.KotlinLogging
import javafx.event.ActionEvent
import javafx.event.EventHandler
import javafx.scene.Scene
import javafx.scene.control.Menu
import javafx.scene.control.MenuItem
import javafx.stage.Stage
import lombok.extern.slf4j.Slf4j
import org.git.desk.loader.FXMLLoaderFactory
import org.kordamp.ikonli.javafx.FontIcon

@Slf4j
class FileMenu : Menu("_File", FontIcon("mdi2f-folder")) {

  private val logger = KotlinLogging.logger { }
  private val addAccount: MenuItem = IconMenuItem("Add Account", "mdi2f-folder-account")
  private val addRepository: MenuItem = IconMenuItem("Add Repository", "mdi2s-source-repository") // 这里可以换别的图标

  init {
    init()
  }

  private fun init() {
    addAccount.onAction = EventHandler { _: ActionEvent? -> openForm("AddAccountForm.fxml", "Add Account") }
    addRepository.onAction = EventHandler { _: ActionEvent? ->
      openForm(
        "AddRepositoryForm.fxml",
        "Add Repository"
      )
    }
    this.items.addAll(addAccount, addRepository)
  }

  private fun openForm(fxmlPath: String?, title: String?) {
    try {
      val root = FXMLLoaderFactory.load(fxmlPath)
      val stage = Stage()
      stage.title = title
      stage.setScene(Scene(root))
      stage.show()
    } catch (e: Exception) {
      logger.atError {
        message = "Failed to open form: $fxmlPath"
        cause = e
      }
    }
  }
}
